subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/AttributeMatchingExplanationBuilder.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.internal.component.model;
+
+import org.gradle.api.attributes.Attribute;
+import org.gradle.api.attributes.HasAttributes;
+import org.gradle.api.internal.attributes.AttributeContainerInternal;
+import org.gradle.api.internal.attributes.AttributeValue;
+
+import java.util.Collection;
+
+public interface AttributeMatchingExplanationBuilder {
+    AttributeMatchingExplanationBuilder NO_OP = new AttributeMatchingExplanationBuilder() {
```

Comment:
:o: Replace with Lambda `AttributeMatchingExplanationBuilder NO_OP = () -> true;`

Reply:
We can't, it's not a SAM-type ;)

Reply:
Doesn't it work because there are defaults for all other methods? IDEA is actually suggesting me this change.

Reply:
Actually you're right it works because there's a single abstract method left to implement.